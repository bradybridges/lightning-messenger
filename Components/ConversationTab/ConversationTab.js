import React from 'react';
import {
  TouchableOpacity,
  Text,
  StyleSheet,
  Dimensions,
} from 'react-native';
import NotificationBadge from '../NotificationBadge/NotificationBadge';
import * as Constants from '../../Constants/Constants';

const ConversationTab = ({
  from,
  time,
  updateSelectedConversation,
  handleConversationTabLongPress,
  newMessageCount,
}) => (
  <TouchableOpacity style={styles.container} onPress={() => updateSelectedConversation(from)} onLongPress={() => handleConversationTabLongPress(from)}>
    <Text style={styles.text}>{from}</Text>
    { newMessageCount === 0 && <Text style={styles.text}>{time}</Text> }
    { newMessageCount > 0 && <NotificationBadge newMessageCount={newMessageCount} /> }
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  container: {
    height: Dimensions.get('window').height * 0.125,
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
    borderBottomColor: Constants.tertiaryBgColor,
    borderBottomWidth: 1,
    paddingHorizontal: Constants.midMarginPadding,
  },
  text: {
    fontSize: 18,
    color: 'white',
    fontFamily: 'exo-regular',
  },
});

export default ConversationTab;
