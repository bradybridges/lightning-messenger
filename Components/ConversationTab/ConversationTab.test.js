import React from 'react';
import { shallow } from 'enzyme';
import ConversationTab from './ConversationTab';

describe('ConversationTab', () => {
  let wrapper;
  const mockUpdateSelectedConversation = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <ConversationTab
        from="sender"
        time="12:00 PM"
        updateSelectedConversation={mockUpdateSelectedConversation}
      />,
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('when pressed the conversation tab should call updateSelectedConversation', () => {
    wrapper.simulate('press');
    expect(mockUpdateSelectedConversation).toHaveBeenCalledWith('sender');
  });
});
