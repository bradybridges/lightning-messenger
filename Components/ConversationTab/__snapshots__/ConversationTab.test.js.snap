// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationTab should match snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#6E8F76",
      "borderBottomWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "height": 166.75,
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 18,
      }
    }
  >
    sender
  </Text>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 18,
      }
    }
  >
    12:00 PM
  </Text>
</TouchableOpacity>
`;
